oaipmhTemplates/dha_RepoObject.xml

```diff
@@ -0,0 +1,154 @@
+    xmlns:skos="&skos;" xmlns:dc="&dc;" xmlns:dct="&dct;" xmlns:foaf="&foaf;" xmlns:dha="&dha;" xmlns:premis="&premis;">
+
+    <!--
+    ////////////////////////////////////////////////////////////////////////////
+    //
+    // Mapping for RepoObject
+    //
+    ////////////////////////////////////////////////////////////////////////////
+    -->
+
+    <dha:RepoObject> <!-- Q: how do I include rdf:about with URI of the Resource? -->
+
+        <dha:hasMetadataDescription rdf:parseType="Resource">
+            <dc:title count="*" lang="true" val="/acdh:hasTitle"></dc:title>
+            <dc:identifier count="*" val="@acdh:hasIdentifier/acdh:hasTitle"></dc:identifier>
+            <dc:creator count="*" val="@acdh:hasCreator/acdh:hasTitle"></dc:creator>
+            <dc:contributor count="*" val="@acdh:hasContributor/acdh:hasTitle"></dc:contributor>
+        </dha:hasMetadataDescription>
+
+        <dha:basedOn> <!-- Q: is it possible to create as many elements for this property as the values for ^acdh:isSourceOf? -->
+            <dha:Object> <!-- Q: Actually, it's not sure that this is a dha:Object, and it may well be a dha:Collection. Is it possible to add some sort of if-condition? If the object is a Collection, then print dha:Collection; if it is a Resource, then print dha:Object -->
+              <!-- Q: A rdf:about attribute should be added here too; see above -->
+              <!-- The following part can be avoided:
+              <dha:hasMetadataDescription rdf:parseType="Resource">
+                  <dc:title count="*" lang="true" val="@^acdh:isSourceOf/acdh:hasTitle"></dc:title>
+                  <dc:identifier count="*" val="@^acdh:isSourceOf/acdh:hasIdentifier/acdh:hasTitle"></dc:identifier>
+                  <dc:creator count="*" val="@^acdh:isSourceOf/acdh:hasCreator/acdh:hasTitle"></dc:creator>
+                  <dc:contributor count="*" val="@^acdh:isSourceOf/acdh:hasContributor/acdh:hasTitle"></dc:contributor>
+              </dha:hasMetadataDescription>
+            -->
+            </dha:Object>
+        </dha:basedOn>
+
+        <dha:hasPreservationDescription rdf:parseType="Resource">
+            <premis:hasIdentifier count="*" val="/acdh:hasPid"></premis:hasIdentifier>
+            <dha:hasAccessRights count="*" val="@acdh:hasAccessRestriction/hasIdentifier"></dha:hasAccessRights> <!-- Q: I need an identifier that is just a xsd:anyURI here, while hasIdentifier is an object property in ARCHE. How can I convert the value to a literal? -->
+            <dha:hasLicense count="*" val="/acdh:hasLicense"></dha:hasLicense> <!-- same here -->
+        </dha:hasPreservationDescription>
+
+        <dha:hasDisseminationDescription rdf:parseType="Resource"> <!-- Q: Would it be possible to list all Dissemination Services and include each of them as separate objects, by repeating the template  in the element <dha:hasDisseminationService>? -->
```

Comment:
No but yes but with some limitations ;-)

* No, there is not such option at the moment
* But yes, we can imagine adding a special `val="{someSpecialValue}"` syntax which would return an URI of all matching dissemination services so combining with `count="*"` and `ComponentId="templateForADissServ"` it would do the job
* But with some limitations as matching dissemination services is quite time consuming. I mean it's not much for a single resource (let's say 0.5s) but if you try to do it for all resources in the repository (currently around 80k) you end up with a lot (11 hours)

Reply:
Ah ok, I understand... quite a difficult choice, because I imagine that when the harvester requests all the resources it will be quite of an overload... what is best to do in these cases? Is there some workaround to limit this?